core/router.ts

```diff
@@ -150,22 +150,30 @@ onEnvironment(({ baseRouter }) => {
             }
           }
 
-          if (page.params.since) {
-            if (isNumber(page.params.since)) {
-              let since =
-                typeof page.params.since === 'number'
-                  ? page.params.since
-                  : parseInt(page.params.since)
+          if (page.params.currentPage) {
```

Comment:
There is a problem with `currentPage` instead of `since`: if user press `Resfresh` the list will start to jump on every new post loaded. It is OK for the first page, but very strange for others.

Right now we extract all posts from storage. What do you think of keeping `since`?

Reply:
Sure, will revert back to `since`.

There were 2 main reasons for the introduction of `currentPage` and `totalPages`:
1. To make the pagination-bar component possible (e.g. you need to have 10 bars if 10 is the total num of pages and you need 3rd bar marked as active if the user is on page 3)
2. To make clicking on the pagination-bar possible. `since` is great for `next-page` usage, but what if the user wants to click on page 5 from page 1?

For 2. we need the ref to `currentPage` to know which part of the `posts` to slice and if we are already slicing based on `currentPage` we don't have a need for `since` so I decided to remove it. 

However, there's another option - where we can keep `since` and navigate based on it but calculate it from `currentPage` and the `page` clicked upon. 
That initially seemed redundant (keeping everything - `since`, `currentPage`, `totalPages`, etc) so I decided to move `since` out completely and just stick with the `currentPage` implementation. 

Unfortunately, I didn't know about the `Refresh` case :/

Reply:
I overslept with this idea and seems like your `currentPage `approach is the better option right now.

I am planning to refactor the data system a lot. This is why before this refactoring we should not spend too much effort of pagination (since we will need to rewrite it).

So, let’s keep it and think about refresh case later. Can I just ask you to rename `currentPage` to `page` to keep the short-name code style?

Reply:
> I overslept with this idea and seems like your `currentPage `approach is the better option right now.
I mean I just didn't see a way how to support "jump to a specific page" functionality with `since` and without `currentPage`, maybe I'm missing something.

> Can I just ask you to rename `currentPage` to `page` to keep the short-name code style?
Great that you brought it up, I wanted to name it `page` initially, 
Wasn't sure how you would like the `page.params.page` so I switched it to `currentPage`.
Will change it :grin:

Reply:
> I mean I just didn't see a way to support "jump to a specific page" functionality with since and without currentPage, maybe I'm missing something.

I don’t see a way right now too